test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
> I would suggest that if you want to make wait.Poll better... that we make wait.Poll better.

yeah but i don't.  there are a variety of ways people want pollers to behave (the poll immediately vs sleep first is one thing, how to handle errors (abort vs continue retrying) is another.

defining, writing, maintaining and trying to get everyone to consistently use the uber-poller library of function isn't worth it for 5 lines of code that took me 30 seconds to write.